e2e/src/usecases/SecureSend.js

```diff
@@ -10,83 +10,152 @@ import {
+      await waitFor(element(by.id('cUSDTouchable')))
+        .toBeVisible()
+        .withTimeout(30000)
+      await element(by.id('cUSDTouchable')).tap()
+
+      // Enter the amount and review
+      await inputNumberKeypad(AMOUNT_TO_SEND)
+      await element(by.id('Review')).tap()
+
+      // Use the last digits of the account to confirm the sender.
+      await waitFor(element(by.id('confirmAccountButton')))
+        .toBeVisible()
+        .withTimeout(30000)
+      await element(by.id('confirmAccountButton')).tap()
+      // TODO: test case for AddressValidationType.PARTIAL but relies on mapping phone number to another address with unique last 4 digits
+      // for (let index = 0; index < 4; index++) {
+      //   const character = LAST_ACCOUNT_CHARACTERS[index]
+      //   await element(by.id(`SingleDigitInput/digit${index}`)).replaceText(character)
+      // }
+      await element(by.id('ValidateRecipientAccount/TextInput')).replaceText(
+        SAMPLE_WALLET_ADDRESS_VERIFIED_2
+      )
+
+      // Scroll to see submit button
+      await scrollIntoView('Submit', 'KeyboardAwareScrollView', 50)
+      await element(by.id('ConfirmAccountButton')).tap()
+
+      // Write a comment.
+      await addComment(randomContent)
+
+      // Confirm and input PIN if necessary.
+      await element(by.id('ConfirmButton')).tap()
+      await enterPinUiIfNecessary()
+
+      // Return to home screen.
+      await waitFor(element(by.id('HomeAction-Send')))
+        .toBeVisible()
+        .withTimeout(30 * 1000)
+
+      await waitFor(element(by.text(`${randomContent}`)))
+        .toBeVisible()
+        .withTimeout(60000)
+    })
+  })
+
+  describe('Secure send flow with phone number lookup (new flow)', () => {
+    beforeAll(async () => {
+      // uninstall the app to remove secure send mapping
+      await device.uninstallApp()
+      await device.installApp()
+      await launchApp({
+        newInstance: true,
+        permissions: { notifications: 'YES', contacts: 'YES' },
+        launchArgs: {
+          statsigGateOverrides: `use_new_send_flow=true,use_viem_for_send=true`,
+        },
       })
-    }
-  )
+      await quickOnboarding(SAMPLE_BACKUP_KEY_VERIFIED)
+    })
+
+    it('Send cUSD to phone number with multiple mappings', async () => {
+      let randomContent = faker.lorem.words()
+      await waitForElementByIdAndTap('HomeAction-Send', 30000)
+      await waitForElementByIdAndTap('SendSelectRecipientSearchInput', 3000)
+      await element(by.id('SendSelectRecipientSearchInput')).replaceText(VERIFIED_PHONE_NUMBER)
+      await element(by.id('RecipientItem')).tap()
+
+      await waitForElementByIdAndTap('SendOrInviteButton', 30000)
+
+      // Use the last digits of the account to confirm the sender.
+      await waitForElementByIdAndTap('confirmAccountButton', 30000)
+      // TODO: test case for AddressValidationType.PARTIAL but relies on mapping phone number to another address with unique last 4 digits
```

Comment:
is the problem here that there are two secure send flows, one for when the recipient has 2+ addresses verified with the same PN that share the last 4 digits, and one where the 2+ addresses have different last 4 digits?

Reply:
seems like the case where they have the same last 4 digits would be more rare.. is that what's tested for here?

Reply:
I think it's fine to choose only one of those cases for an e2e test and leave the other one, but the one chosen should be the more common case, ideally

Reply:
if it's much harder to test the other case for some reason, punting might be ok, but it should be put into a linear issue. this todo is not going to get noticed otherwise for a long time probly.. this file doesn't get a lot of foot traffic

Reply:
it's ok to leave a todo if you plan to deal with it in the same issue you are currently working on, but if that's the case, it's good to leave some indication of that, like `// TODO(ACT-XYZ): foo bar`